### PR TITLE
Alerting: Refactoring to api_prometheus.go.

### PR DESCRIPTION
**Pull request by stevesg**
Refactors the code which generates the rule and alert status responses such
that the functions can be reused by other parts of the codebase.

### CODE REVIEW

**Review by gotjosh — APPROVED** [review]
LGTM.

I believe @grobinson-grafana changed this are recently and it had a fairly decent coverage - code LGTM and tests pass.

I think you should get his review as well.

File: pkg/services/ngalert/api/api_prometheus.go [hunk 1]
```diff
@@ -69,7 +86,7 @@ func (srv PrometheusSrv) RouteGetAlertStatuses(c *contextmodel.ReqContext) respo
 		})
 	}
 
-	return response.JSON(http.StatusOK, alertResponse)
+	return data
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/api/api_prometheus.go:89]: All code above this line I'm not familiar with I'm afraid, so just wanted to check that you've tested this and made sure it works as it did before?

**Review by grobinson-grafana — APPROVED** [review]
Seems OK to me! I've written quite a few tests for this so I would have expected one of them to break if there was a regression here 🙂

**Comment by grobinson-grafana** [comment]
@stevesg

**Comment by stevesg** [comment]
On hold pending more changes

**Pull request closed without merging**